editor/plugins/material_editor_plugin.cpp

```diff
@@ -56,8 +56,14 @@ void MaterialEditor::gui_input(const Ref<InputEvent> &p_event) {
 	if (mm.is_valid() && (mm->get_button_mask().has_flag(MouseButtonMask::LEFT))) {
 		rot.x -= mm->get_relative().y * 0.01;
 		rot.y -= mm->get_relative().x * 0.01;
-
-		rot.x = CLAMP(rot.x, -Math_PI / 2, Math_PI / 2);
+		if (quad_instance->is_visible()) {
+			// Clamp rotation so the quad is always visible.
+			// Magic numbers so that the range is a bit less than 90 degrees in either direction.
```

Comment:
You can make it less magic by doing
```suggestion
			const float limit = Math::deg_to_rad(80);
```

Reply:
Makes sense. I think I'll revisit how all the rotations are handled once the behavior is finalized. Right now it's a mix of radians and degrees, converted back and forth. It can definitely be cleaned up.

Reply:
Made this change, but still going to revisit how rotation is handled later.